### PR TITLE
Add sketches of edit-assignment workflow starting point to pattern library

### PR DESCRIPTION
**Pull request by lyzadanger**
This PR adds a new page in the LMS pattern library to hold sketches of UI options for the edit-assignment workflow. It:

* Captures the existing UI (roughly) for the configure-assignment workflow (i.e. configuring a new assignment)
* Provides some sketches of a possible initial view that an instructor would see after clicking into the edit-assignment workflow from a launched assignment (via the "Edit" button in the instructor toolbar).
* Gives examples to look at when we next discuss the project.

It does not:

* Address behavior in the edit-assignment workflow (yet)
* Provide cleaned-up implementation or full documentation of these interfaces

I want to keep these iterations tight so I don't end up delving too long into UI exploration at any one time. This is purposely in the "messy" stage right now.

To see the sketches, fire up your dev server and go to http://localhost:8001/ui-playground/edit-assignment

Partial screenshots:

image

image

Part of https://github.com/hypothesis/lms/issues/5141
Part of https://github.com/hypothesis/lms/issues/5142

### CODE REVIEW

**Comment by robertknight** [comment]
Unrelated to the UI sketches, it feels like there is too much variety of font families, weights and sizes here. There are additional variations as you scroll down.

UI playground fonts

File: lms/static/scripts/ui-playground/components/EditAssignment.tsx [hunk 1]
```diff
@@ -0,0 +1,348 @@
...
+            The assignment-configuration UI was originally designed under the
+            assumption that it would be followed linearly for an assignment that
+            had not yet been configured. With the introduction of support to
+            edit existing assignment configuration, the UI needs to be
+            revisited.
+          </p>
+        </>
+      }
+    >
+      <Library.Section
```
Comment (robertknight) [inline lms/static/scripts/ui-playground/components/EditAssignment.tsx:100]: Something that might be a little unclear when a new person reads these pages is how the various `Library` components which were originally designed for a hierarchy of `Component group => Component => Usage => Example` map to a hierarchy for workflows / screens. Maybe we could create some aliases or something.

File: lms/static/scripts/ui-playground/components/EditAssignment.tsx [hunk 2]
```diff
@@ -0,0 +1,348 @@
...
+  CardActions,
+  CardContent,
+  Checkbox,
+  LinkButton,
+} from '@hypothesis/frontend-shared/lib/next';
+import Library from '@hypothesis/frontend-shared/lib/pattern-library/components/Library';
+import classnames from 'classnames';
+import type { ComponentChildren } from 'preact';
+
+function PickerCard({
```
Comment (robertknight) [inline lms/static/scripts/ui-playground/components/EditAssignment.tsx:14]: Are you planning on sharing components between the ui-playground and the actual app in future, or intending to keep them separate?

**Review by robertknight — APPROVED** [review]
OK, let's get these "sketches" landed and we can iterate from there.

**Comment by lyzadanger** [comment]
> Unrelated to the UI sketches, it feels like there is too much variety of font families, weights and sizes here. There are additional variations as you scroll down.

Oh believe me I agree. It would be pure luxury to carve out some time to make it better, and also ditto on the structure of the section-ish components themselves.

**Pull request merged**